frontend/src/App.tsx

```diff
@@ -24,58 +24,63 @@ import SubmitProperty from 'pages/SubmitProperty';
 import LoadingBar from 'react-redux-loading-bar';
 import ErrorBoundary from 'react-error-boundary';
 import ErrorModal from 'components/common/ErrorModal';
+import { AuthStateContext, IAuthState } from 'contexts/authStateContext';
 
 export const store = configureStore();
 
 const App = () => {
   const keycloakWrapper = useKeycloakWrapper();
   const keycloak = keycloakWrapper.obj;
-  const [keycloakUserLoaded, setkeycloakUserLoaded] = useState({});
   const dispatch = useDispatch();
 
   useEffect(() => {
     if (keycloak?.authenticated) {
       dispatch(getActivateUserAction());
       dispatch(getFetchLookupCodeAction());
     }
-    keycloak?.loadUserInfo().then(() => {
-      setkeycloakUserLoaded(true);
-    });
   }, []);
 
-  const isInitialized = () => {
-    return keycloak && keycloakUserLoaded;
-  };
+  return (
```

Comment:
Does this allow for an unauthorized user (a true guest) have access to public routes?

Reply:
From the if statement it looks like we'll get the following behaviour:
authenticated + userinfo = display app (this is our happy path)
not authenticated + no userinfo = display app (this is our not-logged-in user path)
authenticated +  no userinfo = display loading bar (presumably this is because the userinfo request is still loading)
not authenticated + userinfo = display app? This is a weird case, not sure if it could ever come up, but it doesn't seem right. I think the userinfo request will fail if the user is not authenticated.

as a result, could we simplify this to:
(!context.ready || (context.ready && context.authenticated && !context.userInfo))

Reply:
@devinleighsmith you right, a result of moving code around. I will make the change and test it